app/_components/SubmitGuessButton.tsx

```diff
@@ -0,0 +1,111 @@
+import { useState, useEffect } from "react";
+import { Button } from "./ui/button";
+import { Pin } from "../_utils/global";
+import {
+  Coordinates,
+  GetDistanceFromCoordinatesToMeters,
+} from "../_utils/coordinateMath";
+
+
+interface SubmitGuessButtonProps {
+  pins: Pin[];
+}
+
+/**
+ * TODO: Manage Button states based on closest pin's search radius.
+ * This can done by constantly checking the user coordinates.
+ * Manage how frequent this call is made.
+ **/
+
+function SubmitGuessButton({
+  pins,
+}: SubmitGuessButtonProps): React.JSX.Element {
+  const [trackingPin, setTrackingPin] = useState<Pin | null>(null);
+  const [distanceToPin, setDistanceToPin] = useState<number>(0);
+  const [isActiveState, setIsActiveState] = useState<boolean>(false);
+
+  useEffect(() => {
+    const id = navigator.geolocation.watchPosition((position) => {
+      handleTrackingPinAndDistanceToPin(position.coords);
+    });
+    return () => navigator.geolocation.clearWatch(id);
+  });
```

Comment:
Is this logic being used in other components?

If so, it would be best to convert it into a custom hook that takes in a callback (in this case, `handleTrackingPinandDistanceToPin`) so that we can reuse this code in other components.

https://react.dev/learn/reusing-logic-with-custom-hooks